contracts/peg/BasePSM.sol

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {IPCVDeposit, PCVDeposit} from "./../pcv/PCVDeposit.sol";
+import {IBasePSM} from "./IBasePSM.sol";
+import {OracleRef, Decimal, SafeCast} from "./../refs/OracleRef.sol";
+import {Constants} from "../Constants.sol";
+import {SafeERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {TribeRoles} from "../core/TribeRoles.sol";
+
+abstract contract BasePSM is IBasePSM, OracleRef, PCVDeposit {
+    using Decimal for Decimal.D256;
+    using SafeCast for *;
+    using SafeERC20 for IERC20;
+
+    /// @notice the token this PSM will exchange for VOLT
+    IERC20 public immutable override underlyingToken;
+
+    /// @notice constructor
+    /// @param params PSM constructor parameter struct
+    constructor(OracleParams memory params, IERC20 _underlyingToken)
+        OracleRef(
+            params.coreAddress,
+            params.oracleAddress,
+            params.backupOracle,
+            params.decimalsNormalizer,
+            params.doInvert
+        )
+    {
+        underlyingToken = _underlyingToken;
+
+        _setContractAdminRole(TribeRoles.PSM_ADMIN_ROLE);
```

Comment:
let's remove this use of PSM_ADMIN_ROLE as we want to deprecate as many unnecessary roles as possible